Casks/s/stack.rb

```diff
@@ -1,15 +1,19 @@
 cask "stack" do
-  version "2.11.1-20220906"
-  sha256 "d5b92e5f0070fbd1f92c36be58e0c979efcceb79b685dd7e665ebf165d70a82c"
+  arch arm: "arm64", intel: "x86_64"
 
-  url "https://filehosting-client.transip.nl/packages/stack/v#{version}/macos/stack-v#{version}.dmg"
+  version "5.3.1-20240731"
+  sha256 arm:   "bd8efc4954f7f6e14f0b55ed19cd462008943cce5d7a9dd3c7b8362a283e8d0f",
+         intel: "11788fd9d523bfa215b2bed2478717d7f58f096f31bdadd5dc415a9c1081aa45"
+
+  url "https://filehosting-client.transip.nl/packages/stack/v#{version}/macos/stack-v#{version}.#{arch}.dmg"
   name "STACK"
   desc "Personal online hard drive to store, view and share files"
   homepage "https://www.transip.nl/stack/"
 
   livecheck do
-    url "https://filehosting-client.transip.nl/packages/stack/"
-    regex(%r{href=["'](?:.*?/)?v?(2(?:[.-]\d+)+)["' >]}i)
+    url "https://filehosting-client.transip.nl/packages/stack-macos-latest.#{arch}.dmg"
+    regex(/stack[._-]v?(\d(?:[._-]\d+)+)\.#{arch}\.dmg/i)
```

Comment:
```suggestion
    regex(/stack[._-]v?(\d(?:[.-]\d+)+)[._-]#{arch}\.dmg/i)
```

When handling a version like `5.3.1-20240731`, I simply replace `\.` with `[.-]` in the repeating numeric part. The idea is to use only the known delimiters in that part, as `[._-]` may end up matching more than we want if the filename format changes in the future.

On the flip side, we should use `[._-]` instead of `\.` in the arch suffix, as we want the regex to continue matching if the delimiter changes to `-` or `_` in the future.